src/config.rs

```diff
@@ -74,20 +74,16 @@ impl Default for Config {
 
 impl Config {
     /// Validate the configuration integrity.
-    pub async fn validate(&self) -> Result<()> {
+    pub fn validate(&self) -> Result<()> {
         if !self.runtime().exists() {
             bail!("runtime path '{}' does not exist", self.runtime().display())
         }
 
         if self.socket().exists() {
-            fs::remove_file(self.socket())
-                .await
-                .context("remove existing socket file")?;
+            fs::remove_file(self.socket())?;
```

Comment:
because we don't want to start tokio until after the fork, but we want the information from the config before the fork. so we need to process the config fields synchronously

Reply:
Yeah that's fine. But the context adds additional information to the error message to better trace where the error is coming from. I think we can keep it for now, but later on we probably wanna add more `.context("…")` descriptions to our error messages to not trace "no such file or directory" message through the whole stack.